tests/e2e/helpers/desktopSearch.js

```diff
@@ -0,0 +1,69 @@
+import { ClientFunction, RequestMock } from 'testcafe';
+import { searchLink, mockResponseFound, mockResponseNotFound } from './mockSearch';
+
+
+export function runDesktopSearchTests(br) {
+  const testTextFound = 'theory';
+  const testTextNotFound = 'HopefullyNotFoundLongWord';
+
+  //building mock response  for successful and unsuccessful search
+  const mockFound = RequestMock()
+    .onRequestTo(searchLink)
+    .respond(mockResponseFound, 202);
+
+  const mockNotFound = RequestMock()
+    .onRequestTo(searchLink)
+    .respond(mockResponseNotFound, 202);
+
+
+  test
+    .requestHooks(mockFound)('Desktop search - successful search', async t => {
+
+      const { nav } = br;
+      await t.expect(br.shell.visible).ok();
+      await t.expect(br.BRcontainer.visible).ok();
+
+      //assuring that the search bar is enabled
+      await t.expect(nav.desktop.searchBox.exists).ok();
+      await t.expect(nav.desktop.searchBox.visible).ok();
+
+      //testing search for a word found in the book
+      await t
+        .selectText(nav.desktop.searchBox.child('.BRsearchInput'))
+        .pressKey('delete');
+      await t.typeText(nav.desktop.searchBox.child('.BRsearchInput'), testTextFound);
+      await t.click((nav.desktop.searchBox).child('.BRsearchSubmit'));
+      await t.expect(nav.desktop.querySign.exists).ok({timeout:20000});
+      await t.expect(nav.desktop.querySign.child('div').exists).ok();
+      await t.expect(nav.desktop.querySign.child('div').innerText).contains(testTextFound);
+
+      const getPageUrl = ClientFunction(() => window.location.href.toString());
+      await t.expect(getPageUrl()).contains(testTextFound);
+
```

Comment:
Let's also check that the search pins are visible in this case.

Reply:
Already done with querySign (previous issue)! (Sorry for poor choice for variable name)